Deduplicate and update TP instructions

Sjors opened the pull request:
The Template Provider is now a standalone application that connects to Bitcoin Core. Update the instructions to install both.

To avoid having to update the same documentation in ~three~ two places, move them out to helpers:
- one for installing Bitcoin Core v30
- one for installing sv2-tp (from it's new location https://github.com/stratum-mining/sv2-tp)

The first commit just fixes some whitespace.

The second commit introduces a helper that adds support for reusing markdown snippets. It's entirely vibe-coded and maybe there's an easier way?

plebhash commented on app/src/_blog/getting-started.md:
```diff
@@ -188,7 +150,7 @@ Setup the correct CPUMiner for your OS:
 - Or compile it from [https://github.com/pooler/cpuminer](https://github.com/pooler/cpuminer)
 
 On the CPUMiner directory:
- 
```
there's many changes in the diff that seem to be just adding/removing empty spaces and linebreaks

can we clean those up?

Sjors replied:
Those are in a separate commit: 52873fde18ce4dcf8b067afa6b9f523f5f000277

I can drop that commit if needed.

xyephy reviewed (commented):
everything else looks good apart from the two comments I've left in -app/src/_shared/run-template-provider.md.

Sjors commented:
@xyephy thanks, fixed the paths.

lucasbalieiro commented on app/routes/blog.$slug.tsx:
I think we shouldn't change the behavior of this file and the one from `app/src/_blog/getting-started.md`

Because this is probably some leftover from this PR here: https://github.com/stratum-mining/stratumprotocol.org/pull/277

The `app/src/_blog/getting-started.md` should be removed because its not needed anymore.

Sjors replied:
I opened #288 to drop that. Let's merge that first so this PR doesn't get too complicated.

Sjors commented:
Rebased after #288 so now we're down from 3x to 2x the same text. Added the missing -`testnet4`.

I dropped the changes to `app/routes/blog.$slug.tsx` from cdbc0448dd1ce608b37392d66ec5e7202bc8b60b, since we're no longer editing the blog part. That reduces the amount of vibe coded stuff.

GitGab19 commented on app/src/_developers/getting-started.md:
```diff
@@ -212,47 +174,9 @@ Another parameter you can tune is `min_individual_miner_hashrate` in `tproxy-con
-This way new templates are constructed every 20 seconds (taking the most profitable txs from the mempool) and they are sent downstream if new fees collected are more than 1000 sats. 
+{{< include "_shared/run-template-provider.md" >}}
 
 #### Clone the repository
```
```suggestion
#### Clone the SRI repository
```

lucasbalieiro commented:
now, about the replacement code, I think it’s good to go. It took me a bit to understand why it’s recursive, but it makes sense now: one `include` can call another, so we need to keep track of the ones we’ve already parsed using the `Set()` you created. That also prevents infinite recursion, since an include could, by mistake, reference itself and cause some weird behavior.

I also checked the `rehype` ecosystem (the tool we’re using to handle some markdown stuff) to see if there’s anything similar. I did find one, but it’s pretty outdated.

unfortunately, we’ll have to keep parsing these Markdown files ourselves and keep doing these acrobatics.

lucasbalieiro commented:
> Let's log an issue for this @lucasbalieiro on `sv2-apps`, where we can better discuss this without polluting this PR.

Issue being tracked in https://github.com/stratum-mining/sv2-apps/issues/63

GitGab19 reviewed (approved):
Given we solved the issue described in https://github.com/stratum-mining/sv2-apps/issues/63, we can safely merge this PR now.
